bundles/org.openhab.binding.casokitchen/src/main/java/org/openhab/binding/casokitchen/internal/handler/TwoZonesWinecoolerHandler.java

```diff
@@ -0,0 +1,255 @@
+                            case TARGET_TEMPERATURE:
+                                updateState(new ChannelUID(thing.getUID(), group, TARGET_TEMPERATURE),
+                                        QuantityType.valueOf(result.targetTemperature2, SIUnits.CELSIUS));
+                                break;
+                        }
+                        break;
+                }
+            });
+        } else if (LIGHT.equals(channelUID.getIdWithoutGroup())) {
+            LightRequest lr = new LightRequest();
+            lr.technicalDeviceId = configuration.deviceId;
+            if (command instanceof OnOffType) {
+                lr.lightOn = OnOffType.ON.equals(command);
+                switch (group) {
+                    case GENERIC:
+                        lr.zone = 0;
+                        break;
+                    case TOP:
+                        lr.zone = 1;
+                        break;
+                    case BOTTOM:
+                        lr.zone = 2;
+                        break;
+                }
+                Request req = httpClient.POST(LIGHT_URL);
+                req.header(HttpHeader.CONTENT_TYPE, "application/json");
+                req.header(HTTP_HEADER_API_KEY, configuration.apiKey);
+                req.content(new StringContentProvider(GSON.toJson(lr)));
+                try {
+                    ContentResponse response = req.timeout(60, TimeUnit.SECONDS).send();
+                    int responseStatus = response.getStatus();
+                    if (responseStatus == 200) {
+                        updateState(new ChannelUID(thing.getUID(), group, LIGHT), OnOffType.from(lr.lightOn));
+                    } else {
+                        logger.info("Call to {} responded with status {} reason {}", LIGHT_URL, response.getStatus(),
+                                response.getReason());
+                    }
+                } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                    logger.debug("Call to {} failed with reason {}", LIGHT_URL, e.getMessage());
+                }
+            }
+        } else {
+            logger.debug("Cannot handle command {}", command);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        refreshJob.ifPresent(job -> {
+            job.cancel(true);
+        });
+    }
+
+    private void dataUpdate() {
+        StatusRequest requestContent = new StatusRequest(configuration.deviceId);
+        Request req = httpClient.POST(STATUS_URL);
+        req.header(HttpHeader.CONTENT_TYPE, "application/json");
+        req.header(HTTP_HEADER_API_KEY, configuration.apiKey);
+        req.content(new StringContentProvider(GSON.toJson(requestContent)));
+        try {
+            ContentResponse contentResponse = req.timeout(60, TimeUnit.SECONDS).send();
+            int responseStatus = contentResponse.getStatus();
+            String contentResult = contentResponse.getContentAsString();
```

Comment:
See other comment about extracting request/response. Maybe extracting it to an additionl class is a biut too much, but atleast extract it to another method.

Reply:
post function extracted